app/models/manageiq/providers/vmware/infra_manager/inventory/saver.rb

```diff
@@ -18,7 +18,12 @@ def save_inventory(persister)
   private
 
   def update_ems_refresh_stats(ems, error: nil)
-    ems.update(:last_refresh_error => error, :last_refresh_date => Time.now.utc)
+    refresh_date = Time.now.utc
+
+    ems.last_refresh_error        = error
+    ems.last_refresh_date         = refresh_date
+    ems.last_refresh_success_date = refresh_date if error.nil?
+    ems.save!
```

Comment:
Interesting - I expected this to be a core method shared by all

Reply:
https://github.com/ManageIQ/manageiq/pull/21888
Since this refreshes completely differently it can't use the same code in core

Reply:
I see - very interesting

Reply:
I think it makes sense to move this method to core and have https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/base_manager/refresher.rb#L63 call `update_ems_refresh_stats(ems)` passing `nil` as the error parameter

Reply:
btw, why does this call `ems.save!` instead of `ems.update!`, as in core?

Reply:
yeah I agree that maybe a shared core method just for this piece to do the updating of these fields would make sense.  Then both core and vmware can run this common method

Reply:
> btw, why does this call ems.save! instead of ems.update!, as in core?

Because this is handling the success and failure cases and I liked the way this read over setting a `last_refresh_success_date` local variable just to one-line an `.update!(*values)`